extensions/wikia/Chat2/ChatBanListSpecial_helper.php

```diff
@@ -221,15 +220,14 @@ private function getUserLinks( $user ) {
 
 		if ( $this->wg->User->isAllowed( 'block' ) && ( !$userIsBlocked ) ) {
 			$links[] = Linker::link(
-				Title::newFromText( "BlockIP/{$user->getName()}", NS_SPECIAL ),
+				SpecialPage::getSafeTitleFor( 'Block', $username ),
 				$this->wg->Lang->ucfirst( wfMsg( 'blocklink' ) )
 			);
 		}
 		if ( $this->wg->User->isAllowed( 'userrights' ) && ( !$userIsBlocked ) ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)